src/Contracts/AbstractCreateZaakAction.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OWC\Zaaksysteem\Contracts;
+
+use DateTime;
+use Exception;
+use OWC\Zaaksysteem\Contracts\Client;
+use OWC\Zaaksysteem\Endpoints\Filter\RoltypenFilter;
+use OWC\Zaaksysteem\Entities\Rol;
+use OWC\Zaaksysteem\Entities\Zaak;
+use OWC\Zaaksysteem\Entities\Zaakeigenschap;
+use OWC\Zaaksysteem\Entities\Zaaktype;
+use OWC\Zaaksysteem\Foundation\Plugin;
+use OWC\Zaaksysteem\Http\Errors\BadRequestError;
+use OWC\Zaaksysteem\Support\PagedCollection;
+use OWC\Zaaksysteem\Traits\ResolveBSN;
+use OWC\Zaaksysteem\Traits\ZaakTypeByIdentifier;
+
+abstract class AbstractCreateZaakAction
+{
+    use ResolveBSN;
+    use ZaakTypeByIdentifier;
+
+    public const CALLABLE_NAME = '';
+    public const CLIENT_CATALOGI_URL = '';
+    public const CLIENT_ZAKEN_URL = '';
+    public const FORM_SETTING_SUPPLIER_KEY = '';
+
+    protected Plugin $plugin;
+
+    public function __construct(Plugin $plugin)
```

Comment:
Typehint op `Container` i.p.v. `Plugin`? Alle child classes doen alleen maar `$this->plugin->getContainer()`.

Of de desbetreffende `Client` via de `__construct()` er in en dan `$rsin` als verplichte parameter bij de `addZaak()` method. In de `OWC\Zaaksysteem\GravityForms\GravityForms` class krijg je dan zoiets in de `createZaak()` method:

```php
$action = sprintf('OWC\Zaaksysteem\Clients\%s\Actions\CreateZaakAction', $this->supplier);

if (! class_exists($action)) {
    throw new ResourceNotFoundError(sprintf('Class "%s" does not exists. Verify if the selected supplier has an action class', $action));
}

$instance = $this->plugin->getContainer()->get($action);

return $instance->addZaak($entry, $form, $this->plugin->getContainer()->get('rsin'));
```

Ook hier kun je je afvragen waarom de typehint op `Plugin` staat en niet op `Container` :)

Reply:
ik heb een losse ContainerResolver klasse gemaakt die dit afhandelt.